backend/editor/entries.py

```diff
@@ -462,29 +462,38 @@ async def update_nodes(self, label, entry, new_node_keys):
             query.append(f"""\nREMOVE n.{key}\n""")
 
         # Adding normalized tags ids corresponding to entry tags
-        normalised_new_node_keys = {}
-        for keys in new_node_keys.keys():
-            if keys.startswith("tags_"):
-                if "_ids_" not in keys:
-                    keys_language_code = keys.split("_", 1)[1]
+        normalised_new_node = {}
+        for key in new_node.keys():
+            if key.startswith("tags_"):
+                if "_ids_" not in key:
+                    keys_language_code = key.split("_", 1)[1]
                     normalised_value = []
-                    for values in new_node_keys[keys]:
-                        normalised_value.append(normalizer.normalizing(values, keys_language_code))
-                    normalised_new_node_keys[keys] = normalised_value
-                    normalised_new_node_keys["tags_ids_" + keys_language_code] = normalised_value
+                    for value in new_node[key]:
+                        normalised_value.append(normalizer.normalizing(value, keys_language_code))
+                    normalised_new_node[key] = new_node[key]
+                    normalised_new_node["tags_ids_" + keys_language_code] = normalised_value
                 else:
                     pass  # We generate tags_ids, and ignore the one sent
             else:
                 # No need to normalise
-                normalised_new_node_keys[keys] = new_node_keys[keys]
+                normalised_new_node[key] = new_node[key]
 
         # Update keys
-        for key in normalised_new_node_keys.keys():
+        for key in normalised_new_node.keys():
             query.append(f"""\nSET n.{key} = ${key}\n""")
 
+        # Update id if first translation of the main language has changed
+        main_language, id = curr_node["id"].split(":")
+        new_normalised_first_translation = normalizer.normalizing(
+            normalised_new_node["tags_" + main_language][0], main_language
+        )
+        if id != new_normalised_first_translation:
+            normalised_new_node["new_id"] = main_language + ":" + new_normalised_first_translation
+            query.append("""\nSET n.id = $new_id\n""")
```

Comment:
Maybe not in this query, but we may also verify that no other node exists with this new key ? Or we can decide to accept, and report an error in the taxonomy editor, which can be fixed afterward… that might be the best way (tolerance with notification).